readthedocs/search/parsers.py

```diff
@@ -140,21 +140,70 @@ def _parse_sections(self, title, body):
         except Exception as e:
             log.info('Unable to index section', section=str(e))
 
+        # All terms in dls are treated as sections.
+        # After indexing them, we remove them from the body such that their contents aren't
+        # repeated in the following indexing
+        dls = body.css("dl")
+        for dl in dls:
+            # Select all dts with id defined
+            dts = dl.css('dt[id]:not([id=""])')
+
+            # https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dt
+            # multiple <dt> elements in a row indicate several terms that are
+            # all defined by the immediate next <dd> element.
+            for dt in dts:
+                try:
+                    title, _id = self._parse_dt(dt)
+                    # https://developer.mozilla.org/en-US/docs/Web/CSS/General_sibling_combinator
+                    dd = dt.css_first("dt ~ dd")
+                    # We only index a dt with an id attribute and an accompanying dd
+                    if not dd or not _id:
+                        continue
+
+                    # The content of the <dt> section is the content of the accompanying <dd>
+                    content = dd.text()
+                    yield {
+                        "id": _id,
+                        "title": title,
+                        "content": content,
+                    }
+                except Exception:
+                    log.info("Unable to index dt section.", exc_info=True)
+
+            # Remove the <dl> node
+            # There isn't a clear indication if this behavior is DFS or BFS
+            dl.decompose()
```

Comment:
If we aren't sure about the order here, we may just want to duplicate the old logic of making a copy of the node and remove all the children on each iteration.

Reply:
The previous code is doing a copy of the tag before removing its children, so isn't the same as the current code. The current code is relying on the order of the returned elements to remove the correct elements. This order, we don't know if it's DFS or random.